src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs

```diff
@@ -194,15 +196,19 @@ public class DownloadPipelineArtifactTaskV2_0_0 : PipelineArtifactTaskPluginBase
                     }
                 }
 
-                if (!pipelineId.HasValue)
+                if (pipelineId == 0)
                 {
                     if (pipelineVersionToDownload == pipelineVersionToDownloadLatest)
                     {
                         pipelineId = await this.GetPipelineIdAsync(context, pipelineDefinition, pipelineVersionToDownload, projectId.ToString(), tagsInput, resultFilter, null, cancellationToken: token);
                     }
                     else if (pipelineVersionToDownload == pipelineVersionToDownloadSpecific)
                     {
-                        pipelineId = Int32.Parse(userSpecifiedpipelineId);
+                        bool isPipelineIdNum = Int32.TryParse(userSpecifiedRunId, out pipelineId);
+                        if(!isPipelineIdNum)
+                        {
+                            throw new ArgumentException("RunId/PipelineId is not a valid number.");
```

Comment:
A better o/p might be - Pipelineid is not valid - similar to "BuildIdIsNotValid": "Build Id is not valid: {0}",

Reply:
Sounds good.